src/Illuminate/Routing/Redirector.php

```diff
@@ -82,7 +82,13 @@ public function refresh($status = 302, $headers = [])
      */
     public function guest($path, $status = 302, $headers = [], $secure = null)
     {
-        $this->session->put('url.intended', $this->generator->full());
+        $request = $this->generator->getRequest();
+
+        $intended = $request->method() === 'GET' && $request->route() && ! $request->ajax() ?
```

Comment:
Out of curiosity - why is `ajax` specifically excluded?

Reply:
Because we don't want to redirect the user to an AJAX route. We want to only redirect someplace that can be meaningfully loaded up in the browser.

See also `Illuminate\Session\Middleware\StartSession::storeCurrentUrl()`.

Reply:
An unlikely --but possible-- scenario is that the user has an expired session, makes an AJAX request, then manually goes over to the login page to log back in. In that case, the AJAX request could have gotten stored as the intended route, since there'd be nothing to overwrite the value in session.